src/Users/ContactInfo.php

```diff
@@ -0,0 +1,384 @@
+        ];
+        $this->data->phone[] = $phone_obj;
+        return Phone::make($this->client, $phone_obj);
+    }
+
+    /**
+     * Gets the user's preferred phone number, or null if none are preferred
+     * 
+     * @return Phone|null
+     */
+    public function getPreferredPhone()
+    {
+        foreach ($this->data->phone as $phone) {
+            if ($phone->preferred) {
+                return Phone::make($this->client, $phone);
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Remove the preferred flag from all phone numbers
+     */
+    public function unsetPreferredPhone()
+    {
+        foreach ($this->data->phone as $phone) {
+            if ($phone->preferred) {
+                $phone->preferred = false;
+            }
+        }
+    }
+
+    /**
+     * Sets the given phone number as the user's preferred number, adding it as an internal home phone if necessary
+     * 
+     * @param string $phone_number The phone number
+     */
+    public function setPreferredPhone($phone_number)
+    {
+        $this->unsetPreferredPhone();
+        foreach ($this->data->phone as $phone) {
+            if ($phone->phone_number === $phone_number) {
+                $phone->preferred = true;
+                return;
+            }
+        }
+        throw new Exception('Phone number ' . $phone_number . ' not found in user');
+    }
+
+    /**
+     * Removes a phone number from the user
+     * 
+     * @param string The phone number
+     */
+    public function removePhone($phone_number)
+    {
+        foreach ($this->data->phone as $key => $phone) {
+            if ($phone->phone_number === $phone_number) {
+                array_splice($this->data->phone, $key, 1);
+                return;
+            }
+        }
+    }
+
+    /**
+     * Returns an array of all phone numbers associated with the user
+     * 
+     * @return array An array of Phone objects
+     */
+    public function allPhones()
+    {
+        $phones = [];
+        foreach ($this->data->phone as $phone) {
+            $phones[] = Phone::make($this->client, $phone);
+        }
+        return $phones;
+    }
+
+    /**
+     * Gets the user's preferred email address
+     * @return Email The email address
+     */
+    public function getEmail()
+    {
+        if ($this->data->email) {
+            foreach ($this->data->email as $email) {
+                if ($email->preferred) {
+                    return Email::make($this->client, $email);
+                }
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Sets the user's preferred email address, adding a new email address if needed.
+     * @param string $email_address The email address
+     */
+    public function setEmail($email_address)
+    {
+        if ($email_address === $this->getEmail()) {
+            return;
+        }
+        $this->unsetEmail();
+        if ($this->data->email) {
+            foreach ($this->data->email as $email) {
```

Comment:
I think iterator now returns objects rather than strings, so if we are really taking in a `@param` of string, we won't get a true `===`.

Reply:
I changed these to comparisons with the correct property on the objects rather than the objects themselves.